symforce/slam/imu_preintegration/imu_factor.h

```diff
@@ -0,0 +1,76 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/pose3.h>
+
+#include "./preintegrated_imu_measurements.h"
+
+namespace sym {
+
+/**
+ * A Functor storing on-manifold preintegrated IMU measurements between two times and
+ * which computes the local-coordinate difference between the final state (pose and
+ * velocity) and the state you would expect given the initial state and
+ * the IMU measurements.
+ */
+template <typename Scalar>
+class ImuFactor {
+ private:
+  PIMNoCovariance<Scalar> pim_;
+  Eigen::Matrix<Scalar, 9, 9> sqrt_info_;
```

Comment:
I typically lean toward no const here, for the reason you state that it disables the default copy and move when there's otherwise nothing wrong with them

Reply:
Having const member variables makes the compiler generates a throwing move, which can be suboptimal if used with stl containers. (It won't have any effect here since everything is on the stack).

Reply:
Sounds good. I'll leave it as no `const` then.